docs/src/api/apply.md

```diff
@@ -260,13 +253,77 @@ If any of these are blockers for you,
+- Multi-field associative list keys
+  - A key that's composed of two or more fields (e.g. both `port` and `protocol`).
+- Scalar-valued associative lists
+  - A list of scalars (not objects) that should be merged as if the
     scalar values were field names in an object.
+
+## Server-Side Apply
+
+Server-side apply (SSA) is a Kubernetes-native declarative update mechanism that allows clients (e.g., controllers) to send a full object definition to the API server, which then manages field ownership and performs merges.
+
+SSA provides several advantages over client-side apply:
+
+- **Field Ownership Tracking**: The Kubernetes API server tracks which controller or user modified each field, preventing unintended overwrites.
+- **Strategic Merging**: Unlike dynamic apply, SSA applies **strategic merges** even inside CRDs, similar to native Kubernetes resources.
+- **Better Handling of Concurrent Updates**: SSA provides better conflict resolution when multiple controllers modify the same resource.
+
+### How Metacontroller Uses Server-Side Apply
+
+When enabled, Metacontroller will:
+
+- Use the `apply` verb with `server-side=true` when updating managed objects.
+- Allow multiple controllers to modify different fields of the same resource without conflicts.
+- Automatically merge updates to associative lists like `containers`, `volumes`, and `ports` without deleting unexpected changes.
+
+### Example of Server-Side Apply
+
+Instead of performing a full replacement, SSA updates only the fields specified:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-pod
+  labels:
+    app: my-app
+    role: staging # added a label
+spec:
+  containers:
+    - name: nginx
+      image: nginx
+```
+
+When applied using SSA, Kubernetes ensures that fields like `resourceVersion`, `creationTimestamp`, and dynamically added containers (e.g., sidecars) **remain untouched**, unlike dynamic apply which would overwrite lists.
+
+### Comparison: Dynamic Apply vs. Server-Side Apply
+
+| Feature               | Dynamic Apply                       | Server-Side Apply (SSA)           |
+| --------------------- | ----------------------------------- | --------------------------------- |
+| Merging of CRD fields | Uses conventions (e.g., `name` key) | Full strategic merge support      |
+| Field ownership       | Not explicitly tracked              | Kubernetes tracks field ownership |
+| Concurrent updates    | Risk of overwriting fields          | Controlled conflict resolution    |
+| Associative lists     | Convention-based merging            | Kubernetes-native merging         |
+| Performance           | Fast (no API tracking)              | Slightly higher API overhead      |
+
+### Enabling Server-Side Apply
+
+To enable SSA in Metacontroller, configure the controller with:
+
+```sh
+--apply-strategy=server-side-apply
+```
+
+This setting ensures Metacontroller applies resources using Kubernetes-native SSA rather than dynamic apply.
```

Comment:
maybe add a link to https://kubernetes.io/docs/reference/using-api/server-side-apply/ at the end ?